src/Exception/Schema/DataEncode/DataEncodeException.php

```diff
@@ -0,0 +1,31 @@
+<?php
+namespace Shale\Exception\Schema\DataEncode;
+
+use Throwable;
+use Shale\Exception\Schema\SchemaException;
+
+class DataEncodeException extends SchemaException
+{
+    public $problemValue;
+
+    public function __construct(
+        string $message = "",
+        $problemValue = null,
```

Comment:
Do you have specific type for `$problemValue` ?

Reply:
No